frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -193,30 +195,70 @@ type LegacyDataPickerProps = {
   stageIndex: number;
   table: Lib.TableMetadata | Lib.CardMetadata | undefined;
   placeholder: string;
+  canChangeDatabase: boolean;
   hasMetrics: boolean;
   isDisabled: boolean;
   onChange: (tableId: TableId) => void;
 };
 
-function LegacyDataPicker({
+function EmbeddingDataPicker({
   query,
   stageIndex,
   table,
   placeholder,
+  canChangeDatabase,
   hasMetrics,
   isDisabled,
   onChange,
 }: LegacyDataPickerProps) {
+  const { data: dataSourceCountData, isLoading: isDataSourceCountLoading } =
```

Comment:
We'll have to finish this response first, so we could determine if which data picker we should render. See [the decision](https://metaboat.slack.com/archives/C0645JP1W81/p1738608097275359?thread_ts=1738593795.552259&cid=C0645JP1W81) here.